### PR TITLE
Adding vtpm support to nova libvirt driver

### PR DESCRIPTION
**Pull request by paul-emile-element**
Closes-bug: #1798892

Co-Authored-By: Kristine Bujold <kristine.bujold@windriver.com>
Co-Authored-By: Paul-Emile Element <Paul-Emile.Element@windriver.com>
Signed-off-by: Paul-Emile Element <Paul-Emile.Element@windriver.com>

### CODE REVIEW

**Comment by andy-ning** [comment]
Is this a ported commit from Titanium?

**Comment by paul-emile-element** [comment]
> Is this a ported commit from Titanium?

Yes.

**Comment by andy-ning** [comment]
Looks good to me.

**Comment by dtroyer** [comment]
What is the upstream story here?  If this is new code I really want to already know if this is being proposed upstream, if it is a backport or exactly how we avoid adding unaddressed tech debt here.

File: nova/api/openstack/compute/flavors_extraspecs.py [hunk 1]
```diff
@@ -467,7 +467,7 @@ def _validate_min_vcpus(flavor):
     @staticmethod
     def _validate_sw_keys(flavor):
         keys = ['sw:wrs:auto_recovery', 'sw:wrs:guest:heartbeat',
-                'hw:hpet']
+                'sw:wrs:vtpm', 'hw:hpet']
```
[the remark this answers is not in the document]
Comment (paul-emile-element) [reply nova/api/openstack/compute/flavors_extraspecs.py]: This is a back port of a piece of code that should have been included in the initial drop.
The intention is to address the :wrs: references separately.

[the remark this answers is not in the document]
Comment (cbf123) [reply nova/api/openstack/compute/flavors_extraspecs.py]: I think it'd make more sense to switch it to "stx" now.

**Comment by cbf123** [comment]
Dean: the upstream story is that we have a spec under review: https://review.openstack.org/#/c/571111

**Comment by dtroyer** [comment]
Thanks Chris.  I would like to have that in the commit message.

**Comment by paul-emile-element** [comment]
Dean and Chris: I can certainly add the reference to the spec in the commit message.

File: nova/api/openstack/compute/flavors_extraspecs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (brent-rowsell) [reply nova/api/openstack/compute/flavors_extraspecs.py]: Since we are working on upstream this capability' leave it as wrs.

**Pull request merged**